mailpoet/lib/Automation/Engine/Storage/AutomationStatisticsStorage.php

```diff
@@ -76,17 +76,19 @@ private function getStatistics(array $automationIds, int $versionId = null): arr
     return array_combine(array_column($results, 'id'), $results) ?: [];
   }
 
-  private function getStatsQuery(array $automationIds, int $versionId = null, string $status = null): string {
+  private function getStatsQuery(array $automationIds, int $versionId = null, \DateTimeImmutable $after = null, \DateTimeImmutable $before = null, string $status = null): string {
     $table = esc_sql($this->table);
     $placeholders = implode(',', array_fill(0, count($automationIds), '%d'));
     $versionCondition = strval($versionId ? $this->wpdb->prepare('AND version_id = %d', $versionId) : '');
     $statusCondition = strval($status ? $this->wpdb->prepare('AND status = %s', $status) : '');
+    $dateCondition = $after !== null && $before !== null ? strval($this->wpdb->prepare('AND created_at BETWEEN %s AND %s', $after->format('Y-m-d H:i:s'), $before->format('Y-m-d H:i:s'))) : '';
```

Comment:
Minor question: do we not care about queries that are only for `after` or `before`?

Reply:
In the current implementation, we have always both. What do you think, should I prepare for a situation where we only have `$before` or `$after`?